### ISSUE TITLE
Colmena deployment transfers files to remote /nix/store very slowly in v0.4.0. Earlier version works fine.

### ISSUE THREAD

**Issue by bratfizyk** [zhaofengli/colmena#212]
I've set up a simple deployment to a DigitalOcean Droplet, but it's running very slow.
I guess I could have made an error somewhere with the config, but no idea where. I would be happy to hear some hints.

After executing `colmena apply` everything builds fine locally, but once the deployment reaches the stage when it copies files to remote `/nix/store/...` everything slows down to unacceptable state (i.e. after 5h it still hasn't deployed a small application).

It looks like this:
![image](https://github.com/zhaofengli/colmena/assets/10075099/c75beabd-d590-4ea3-a318-6dc3f7238de7)

And then it more or less gets stuck copying some of the packages. To speed this up I cancel the deployment, ssh the machine, execute `nix-store --repair-path /nix/store/<package>` and restart the deployment, but for hundreds of packages this is a bit annoying, at best.

What's surprising, switching `deployment.buildOnTarget` to `true` causes the transfer to go up and complete within 10 minutes.

Has anyone seen anything like this before?

Regarding the flake I use - it's pretty long, so below you can find a truncated version. Happy to post more if this is needed/helpful.

```nix
{
  inputs = {
    nixpkgs.url = "nixpkgs/nixos-23.11";    
    flake-parts.url = "github:hercules-ci/flake-parts";
    haskell-flake.url = "github:srid/haskell-flake";
  };

  outputs = inputs@{ self, nixpkgs, flake-parts, ... }:
    let
      system = "x86_64-linux";
      pkgs = nixpkgs.legacyPackages.${system};
    in
      flake-parts.lib.mkFlake { inherit inputs; } {
        systems = [ system ];
        imports = [
          inputs.haskell-flake.flakeModule
        ];

        flake = {
          colmena = {
            meta = {
              nixpkgs = import inputs.nixpkgs { inherit system; };
            };

            defaults = { pkgs, ... }: {
              environment.systemPackages = with pkgs; [
                curl sqlite htop neovim
              ];
              programs.zsh.enable = true;
            };

            digital-ocean-my-server = {
              imports = [ "${nixpkgs}/nixos/modules/virtualisation/digital-ocean-image.nix" ];
              
              system.stateVersion = "23.11";

              deployment = {
                targetHost = "<correct IP of my server>";
                targetPort = 22;
                # surprisingly, uncommenting this increases massively transfer speed
                #buildOnTarget = true;
              };

              time.timeZone = "Etc/UTC";

              networking = {
                hostName = "my-server";
                firewall = {
                  enable = true;
                  allowedTCPPorts = [ 22 80 443 ];
                  allowPing = true;
                };
              };
     // ....
}
```

**Comment by bratfizyk** [comment]
Update: In the meantime I worked out one more thing: using an older version of colmena (0.3.2 in my case) works perfectly fine and copies all the files very quickly.

I noticed that it uses ssh://... instead of ssh-ng://... . Is it possible that I misconfigured something on my side, which resulted in the ver slow ssh-ng connection?

**Comment by IncredibleLaser** [comment]
I'm having the same issue, and I assume that this was wrongfully closed: https://github.com/NixOS/nix/issues/3372

**Comment by bratfizyk** [comment]
Update from my site: I worked out the the `nix` version I used on the remote machine was pretty old. I created a new NixOS image for a Digital Ocean machine, created a new Droplet using the new image and everything went smooth.

It's described in my thread on NixOS Discourse: https://discourse.nixos.org/t/colmena-deployment-extremely-slow/47546/13

I think it's ok to close this issue, as it's due to an old system configuration.
